Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -217,7 +218,23 @@ NWC24::ErrorCode NetKDRequestDevice::KDDownload(const u16 entry_index,
 
   // Content metadata
   const std::string content_name = m_dl_list.GetVFFContentName(entry_index, subtask_id);
-  const std::string url = m_dl_list.GetDownloadURL(entry_index, subtask_id);
+  std::string url = m_dl_list.GetDownloadURL(entry_index, subtask_id);
+
+  // Reroute to custom server if enabled.
+  const std::vector<std::string> parts = SplitString(url, '/');
+  if (parts.size() < 3)
+  {
+    // Invalid URL
+    LogError(ErrorType::KD_Download, NWC24::WC24_ERR_SERVER);
+    return NWC24::WC24_ERR_SERVER;
+  }
+
+  if (std::optional<std::string> patch =
+          WC24PatchEngine::GetNetworkPatch(parts[2], WC24PatchEngine::IsKD{true}))
+  {
+    const size_t index = url.find(parts[2]);
+    url.replace(index, parts[2].size(), patch.value());
```

Comment:
This is *better* but I feel you could still trick this if you really tried to...

It's probably *fine* given typical input but if you feel motivated you might want to actually extract the host part of the URI based on https://www.rfc-editor.org/rfc/rfc3986 and just target that part.

Reply:
If you *really* want me to I can fix it, otherwise I might just defer it to another time.

Reply:
Eh... like I said, if you feel motivated. You'd have to actively try to abuse this, and to questionable effect.